packages/global-context-store/src/Store.js

```diff
@@ -44,38 +34,50 @@ const Reducer = (state, action) => {
       break;
   }
 
-  newState.debug.actionsFired = logAction(action);
+  newState.__debug.actionsFired = logAction(action);
   return newState;
 };
 const Store = ({ children }) => {
   const [state, dispatch] = useReducer(Reducer, initialState);
 
-  const set = (key, value) => {
-    dispatch({ type: "SET", payload: { key, value } });
-  };
+  const set = React.useCallback(
+    (key, value) => {
+      dispatch({ type: "SET", payload: { key, value } });
+    },
+    [dispatch]
+  );
   const get = React.useCallback(
     (key) => {
       return lodash_get(state, key);
     },
     [state]
   );
-  const invoke = (path, fn) => {
-    dispatch({ type: "INVOKE", payload: { path, fn } });
-  };
-  const unset = (key) => {
-    dispatch({ type: "UNSET", payload: key });
-  };
+  const invoke = React.useCallback(
+    (path, fn) => {
+      dispatch({ type: "INVOKE", payload: { path, fn } });
+    },
+    [dispatch]
+  );
+  const unset = React.useCallback(
+    (key) => {
+      dispatch({ type: "UNSET", payload: key });
+    },
+    [dispatch]
+  );
 
-  const sendRequest = (layer, action) => {
-    if (!get(["layers", layer, "data", "requests"])) {
-      set(["layers", layer, "data", "requests"], []);
-    }
-    invoke(["layers", layer, "data", "requests"], (prev) => [...prev, action]);
-  };
+  const push = React.useCallback(
+    (path, value) => {
+      if (!get(path)) {
+        set(path, []);
+      }
+      invoke(path, (prev) => [...prev, value]);
+    },
+    [get, set, invoke]
+  );
 
   return (
     <Context.Provider
-      value={{ state, dispatch, set, get, invoke, unset, sendRequest }}
+      value={{ state, dispatch, set, get, invoke, unset, push }}
```

Comment:
Generalized what `sendRequest` was, which was basically an array push. Clients can use it as they see fit, which is what `annotation-toolkit` does.